test/adapter/json_api/key_case_test.rb

```diff
@@ -312,187 +294,156 @@ def test_success_document_key_transform_camel_lower
-          resource.errors.add(:title, 'must be longer')
-
-          serializer = ActiveModel::Serializer::ErrorSerializer.new(resource)
-          adapter = ActiveModelSerializers::Adapter::JsonApi.new(serializer)
-
           assert_raises NoMethodError do
-            adapter.serializable_hash(@options)
+            serialize_resource(@error_resource, options(:krazy, serializer: ActiveModel::Serializer::ErrorSerializer))
           end
         end
 
         def test_error_document_key_transform_camel
-          mock_request(:camel)
-
-          resource = ModelWithErrors.new
-          resource.errors.add(:published_at, 'must be in the future')
-          resource.errors.add(:title, 'must be longer')
-
-          serializer = ActiveModel::Serializer::ErrorSerializer.new(resource)
-          adapter = ActiveModelSerializers::Adapter::JsonApi.new(serializer)
-          result = adapter.serializable_hash(@options)
-
-          expected_errors_object =
-            { :Errors =>
+          result = serialize_resource(@error_resource, options(:camel, serializer: ActiveModel::Serializer::ErrorSerializer))
+          expected_errors_object = {
+            Errors:
               [
-                { :Source => { :Pointer => '/data/attributes/published_at' }, :Detail => 'must be in the future' },
-                { :Source => { :Pointer => '/data/attributes/title' }, :Detail => 'must be longer' }
+                {
+                  Source: { Pointer: '/data/attributes/published_at' },
+                  Detail: 'must be in the future'
+                },
+                {
+                  Source: { Pointer: '/data/attributes/title' },
+                  Detail: 'must be longer'
+                }
               ]
           }
           assert_equal expected_errors_object, result
         end
 
         def test_error_document_key_transform_camel_lower
-          mock_request(:camel_lower)
+          result = serialize_resource(@error_resource, options(:camel_lower, serializer: ActiveModel::Serializer::ErrorSerializer))
+          expected_errors_object = {
+            errors: [
+              {
+                source: { pointer: '/data/attributes/published_at' },
+                detail: 'must be in the future'
+              },
+              {
+                source: { pointer: '/data/attributes/title' },
+                detail: 'must be longer'
+              }
+            ]
+          }
+          assert_equal expected_errors_object, result
+        end
 
-          resource = ModelWithErrors.new
-          resource.errors.add(:published_at, 'must be in the future')
-          resource.errors.add(:title, 'must be longer')
+        private
 
-          serializer = ActiveModel::Serializer::ErrorSerializer.new(resource)
-          adapter = ActiveModelSerializers::Adapter::JsonApi.new(serializer)
-          result = adapter.serializable_hash(@options)
+        def options(key_transform = nil, extra_options = {})
```

Comment:
This method replace `mock_request`. That issue `mock_request` was that it sets the `@options` instance variable. Using it this way made us rely on `@options` when serializing and this sometimes created error and bugs when it was not set. Therefore, I think returning the options are easier to handle and understand rather than depending on the instance variable.

I think that changing the method name to `options_with_serialization_context` or `options_with_mock` might make it easier to understand what it does.